rest-servlet/src/main/java/org/hawkular/metrics/restServlet/influx/InfluxSeriesHandler.java

```diff
@@ -435,11 +437,65 @@ private String getColumnName(SelectQueryDefinitions queryDefinitions) {
                     NumberFunctionArgument argument = (NumberFunctionArgument) aggregationFunctionArguments.get(1);
                     retVal = quantil(list, argument.getDoubleValue());
                     break;
+                case TOP:
+                    isMultipleValues = true;
+                    if (!list.isEmpty() && aggregationFunctionArguments.size() > 0) {
+                        int numberOfTopElement = list.size() >
+                            Integer.valueOf(aggregationFunctionArguments.get(0).toString()) ?
+                            Integer.valueOf(aggregationFunctionArguments.get(0).toString()) : list.size();
+                        for(int elementPos =0; elementPos<numberOfTopElement; elementPos++){
+                            retList.add(list.get(elementPos));
+                        }
+                    }
+                    break;
+                case BOTTOM:
+                    isMultipleValues = true;
+                    if (!list.isEmpty() && aggregationFunctionArguments.size() > 0) {
+                        int numberOfBottomElement = list.size() >
+                            Integer.valueOf(aggregationFunctionArguments.get(0).toString()) ?
+                            Integer.valueOf(aggregationFunctionArguments.get(0).toString()) : list.size();
+                        for(int elementPos = (numberOfBottomElement-1); elementPos>=0; elementPos--){
+                            retList.add(list.get(elementPos));
+                        }
+                    }
+                    break;
+                case HISTOGRAM:
+                case MODE:
```

Comment:
I'm not sure how the  `histogram` and `mode` functions behave. The documentation is quite vague:
http://influxdb.com/docs/v0.8/api/aggregate_functions.html

Have you tried to find the answer on the online InfluxDB service?

I'd be fine with leaving those two functions out of the scope for now.